tests/end_to_end/test_client_json.py

```diff
@@ -91,27 +95,127 @@ def verifyParsedOutputsEqual(
+                self.verifyParsedOutputsEqual(
+                    [reference], "references-get", reference.id)
+
+    def testGetReferenceSet(self):
+        for referenceSet in self._client.searchReferenceSets():
+            self.verifyParsedOutputsEqual(
+                [referenceSet], "referencesets-get", referenceSet.id)
+
+    def testGetVariant(self):
+        start = 0
+        end = 1000
+        referenceName = "1"
+        for dataset in self._client.searchDatasets():
+            for variantSet in self._client.searchVariantSets(dataset.id):
+                variants = self._client.searchVariants(
+                    variantSet.id, start=start, end=end,
+                    referenceName=referenceName)
+                for variant in variants:
+                    self.verifyParsedOutputsEqual(
+                        [variant], "variants-get", variant.id)
+
+    @unittest.skip  # No CLI endpoint
+    def testGetVariantSet(self):
+        for dataset in self._client.searchDatasets():
+            for variantSet in self._client.searchVariantSets(dataset.id):
+                self.verifyParsedOutputsEqual(
+                    [variantSet], "variantsets-get", variantSet.id)
+
+    @unittest.skip  # No internal api
+    def testGetCallSet(self):
+        for dataset in self._client.searchDatasets():
+            for variantSet in self._client.searchVariantSets(dataset.id):
+                for callset in self._client.searchCallSets(variantSet.id):
+                    self.verifyParsedOutputsEqual(
+                        [callset], "callsets-get", callset.id)
+
+    def testSearchDatasets(self):
         iterator = self._client.searchDatasets()
         self.verifyParsedOutputsEqual(iterator, "datasets-search")
 
-    def testSearchAllReferenceSets(self):
+    def testSearchReadGroupSets(self):
+        for dataset in self._client.searchDatasets():
+            iterator = self._client.searchReadGroupSets(dataset.id)
+            self.verifyParsedOutputsEqual(
+                iterator, "readgroupsets-search",
+                "--datasetId {}".format(dataset.id))
+
+    def testSearchReads(self):
+        start = 0
+        end = 1000
+        for dataset in self._client.searchDatasets():
+            for readGroupSet in self._client.searchReadGroupSets(dataset.id):
+                for readGroup in readGroupSet.readGroups:
+                    reference = self._client.searchReferences(
+                        referenceSetId=readGroup.referenceSetId).next()
+                    referenceId = reference.id
+                    iterator = self._client.searchReads(
+                        [readGroup.id], referenceId=referenceId,
+                        start=start, end=end)
+                    args = "--start {} --end {} --readGroupIds {}\
+                    --referenceId {}".format(
+                        start, end, readGroup.id, referenceId)
+                    self.verifyParsedOutputsEqual(
```

Comment:
Maybe we could make this
```python
tests_executed += self.verifyParsedOutputsEqual(...)
```
such that verifyParsedOutputsEqual returns the number of items in the supplied iterator?

Then, at the end, we assert that tests_executed > 0. This is a good strong test then I think, and we don't have to worry too much about discovering good regions.